pkg/trace/config/config.go

```diff
@@ -133,6 +136,29 @@ type Tag struct {
 	K, V string
 }
 
+var (
+	isFargate     bool
+	isFargateOnce sync.Once
+)
+
+// IsFargateInstance reports whether we are in a Fargate instance.
+func IsFargateInstance() bool {
+	isFargateOnce.Do(func() {
```

Comment:
You could have also added a flag to the `Agent` struct and fetched the value once in `NewAgent`.  I'll leave style up to you and the other agent devs though.

Reply:
Yeah I went back and forth between that. Ultimately I've put it back into config. Updated the PR.